src/bootstrap/src/core/build_steps/compile.rs

```diff
@@ -1830,13 +1830,24 @@ impl Step for Assemble {
             })
             .collect::<HashSet<_>>();
 
+        let link_std_into_rustc_driver = builder.link_std_into_rustc_driver(target_compiler.host);
         let sysroot = builder.sysroot(target_compiler);
         let rustc_libdir = builder.rustc_libdir(target_compiler);
         t!(fs::create_dir_all(&rustc_libdir));
         let src_libdir = builder.sysroot_libdir(build_compiler, host);
         for f in builder.read_dir(&src_libdir) {
             let filename = f.file_name().into_string().unwrap();
-            if (is_dylib(&filename) || is_debug_info(&filename)) && !proc_macros.contains(&filename)
+
+            // Find `rustc_driver` and exclude `std` for later stages if we're statically
+            // linking `std` into `rustc_driver`.
```

Comment:
It's good that there's a comment now, but does it match what's done here? Maybe something like "only copy dylibs into the libdir if they're dependencies of the driver, if we're dynamically linking to libstd, or if we're building at stage0 for $reason", would that be more accurate?

Reply:
It's not much point to a comment if it's just an exact copy of the code though.

Reply:
I rewrote this comment now.

Reply:
> It's not much point to a comment if it's just an exact copy of the code though.

An exact copy of the code is obviously not what I was asking for…

The new comment is a better explanation, thanks.